### PR TITLE
[ci][test-data/2] serialize bazel event logs into ray_release objects

### PR DESCRIPTION
**Pull request by can-anyscale**
- Serialize bazel event logs into Test and TestResult objects.
- Also allow the tester container to receive team as a param (instead of the normalized docker_tag), since team is needed to construct test objects

Test:
- CI

### CODE REVIEW

File: ci/docker/forge.Dockerfile [hunk 1]
```diff
@@ -40,7 +40,7 @@ apt-get install -y \
 
 # As a convention, we pin all python packages to a specific version. This
 # is to to make sure we can control version upgrades through code changes.
-python -m pip install pip==23.2.1
+python -m pip install pip==23.2.1 cffi==1.16.0
```
Comment (can-anyscale) [inline ci/docker/forge.Dockerfile:43]: cffi which is a dependency of ray-release requires a _cffi_backend interface on the host machine; this force the installation of that interface (https://buildkite.com/ray-project/premerge/builds/16041#018cecda-91b8-4757-8dfe-68a15b10692c/181-254)

File: ci/ray_ci/tester_container.py [hunk 2]
```diff
@@ -113,6 +118,39 @@ def _upload_build_info(self, bazel_log_dir) -> None:
             ]
         )
 
+    def _upload_test_results(self, bazel_log_dir: str) -> None:
+        for test, result in self._get_test_and_results(bazel_log_dir):
+            # TODO(can): upload test results
+            logger.info(
+                f"Updating test object {test.get_name()} with result {result.status}"
```
Comment (aslonnie) [inline ci/ray_ci/tester_container.py:125]: maybe rephrase this log? feels that this line is lying right now.

File: ci/ray_ci/tester_container.py [hunk 3]
```diff
@@ -113,6 +118,39 @@ def _upload_build_info(self, bazel_log_dir) -> None:
...
+        for file in listdir(bazel_log_dir):
+            log = path.join(bazel_log_dir, file)
+            if path.isfile(log) and file.startswith("bazel_log"):
+                bazel_logs.append(log)
+
+        test_and_results = []
+        # Parse bazel logs and print test results
+        for file in bazel_logs:
+            with open(file, "rb") as f:
+                for line in f.readlines():
```
Comment (aslonnie) [inline ci/ray_ci/tester_container.py:142]: you can just `for line in f`, which will read the lines in a streaming fashion.

`readlines()` will read all lines into memory, and it can be huge sometimes.

File: ci/ray_ci/tester_container.py [hunk 4]
```diff
@@ -33,6 +37,7 @@ def __init__(
         used to run tests in a distributed fashion.
         :param shard_ids: The list of shard ids to run. If none, run no shards.
         """
+        self.team = team
```
Comment (aslonnie) [inline ci/ray_ci/tester_container.py:40]: the team info can be parsed from the tag fields of the test log, right? so you do not need to send it inside here?

this is now restricting a test container to one and only one team, which feels unnecessary.

File: ci/ray_ci/tester_container.py (hunk 2 above)
Comment (can-anyscale) [reply ci/ray_ci/tester_container.py:125]: sound good

File: ci/ray_ci/tester_container.py (hunk 3 above)
Comment (can-anyscale) [reply ci/ray_ci/tester_container.py:142]: nice

File: ci/ray_ci/tester_container.py (hunk 4 above)
Comment (can-anyscale) [reply ci/ray_ci/tester_container.py:40]: yes it can be obtained from the log but the info should come from only one source of truth; i intentionally make the `oncall` as the required param in `test_in_docker` so tests are even runnable without a proper owner.

Comment (can-anyscale) [reply ci/ray_ci/tester_container.py:40]: not really, parsing the team tag syntax is the job of `test_in_docker`; if the syntax is changed to ray_team:core, it's the work between `test_in_docker` and bazel rule to deal with it.

Comment (aslonnie) [reply ci/ray_ci/tester_container.py:40]: talked offline, agreed to move this into `run_tests`, so that is is not on the class instance construction path, which is harder to change/expand when required.

File: ci/ray_ci/test_linux_tester_container.py [hunk 5]
```diff
@@ -157,12 +158,12 @@ def _mock_shard_tests(tests: List[str], workers: int, worker_id: int) -> List[st
     ):
         container = LinuxTesterContainer("team", shard_count=2, shard_ids=[0, 1])
         # test_targets are not empty
-        assert container.run_tests(["t1", "t2"], [])
+        assert container.run_tests("team", ["t1", "t2"], [])
```
Comment (aslonnie) [inline ci/ray_ci/test_linux_tester_container.py:161]: nit: rather than calling it a "team" team, how about say "avengers"?

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/test_linux_tester_container.py (hunk 5 above)
Comment (aslonnie) [reply ci/ray_ci/test_linux_tester_container.py:161]: ( or pick your favorite final fantasy ~ )

**Comment by can-anyscale** [comment]
updated team to

File: ci/ray_ci/test_linux_tester_container.py (hunk 5 above)
Comment (aslonnie) [reply ci/ray_ci/test_linux_tester_container.py:161]: :) still on team "team" here?
